### PR TITLE
Audit for error prone isinstance int/float and add lint

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87345

We recently fixed a bug on symbolic-shapes branch where
an isinstance(x, int) test failed when passed a SymIntNode.
To prevent this, I've added a lint for all the codepaths
where we may pass SymInt/SymFloat directly to reject
direct isinstance int/float tests, and instead use one of
the aliases.  The lint rule explains the options.  I then
go and fix all of them.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -929,7 +931,7 @@ def _fill_aten(a: Tensor, value: NumberType) -> Tensor:
 # div prim performs truncation division on integer inputs
 #   and true division for floating and complex inputs
 def _div_aten(a, b):
-    is_integral = isinstance(a, (bool, int)) or (
+    is_integral = isinstance(a, (bool, int, torch.SymIntNode)) or (
```
Comment (bdhirsh) [inline torch/_prims/__init__.py:934]: bleh, I guess it feels pointless to make a dedicated named value for `(bool, int, torch.SymIntNode)`

**Review by bdhirsh — APPROVED** [review]
nice

File: torch/_prims/__init__.py (hunk 1 above)
Comment (ezyang) [reply torch/_prims/__init__.py:934]: actually it's kind of weird the lint is complaining about this site, hmmmm

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**